src/overlays/actors/ovl_En_Nb/z_en_nb.c

```diff
@@ -31,81 +122,676 @@ const ActorInit En_Nb_InitVars = {
+    s32 ret = false;
+
+    if (ActorCutscene_GetCurrentIndex() == 0x7C) {
+        ActorCutscene_Stop(0x7C);
+        ActorCutscene_SetIntentToPlay(index);
+    } else if (ActorCutscene_GetCanPlayNext(index)) {
+        ActorCutscene_StartAndSetUnkLinkFields(index, &this->actor);
+        ret = true;
+    } else {
+        ActorCutscene_SetIntentToPlay(index);
+    }
+
+    return ret;
+}
+
+s16 func_80BC0050(EnNb* this, s32 arg1) {
+    s16 cutscene = this->actor.cutscene;
+    s32 i;
+
+    for (i = 0; i < arg1; i++) {
+        cutscene = ActorCutscene_GetAdditionalCutscene(cutscene);
+    }
+
+    return cutscene;
+}
+
+typedef enum EnNbBehaviour {
+    /* 0 */ ENNB_BEHAVIOUR_0,
+    /* 1 */ ENNB_BEHAVIOUR_1,
+    /* 2 */ ENNB_BEHAVIOUR_2,
+    /* 3 */ ENNB_BEHAVIOUR_3,
+    /* 4 */ ENNB_BEHAVIOUR_4,
+    /* 5 */ ENNB_BEHAVIOUR_5,
+    /* 6 */ ENNB_BEHAVIOUR_6,
+    /* 7 */ ENNB_BEHAVIOUR_7,
+    /* 8 */ ENNB_BEHAVIOUR_8,
+    /* 9 */ ENNB_BEHAVIOUR_9
+} EnNbBehaviour;
+
+s32 func_80BC00AC(Actor* thisx, PlayState* play) {
+    EnNb* this = THIS;
+    s16 cutscene = func_80BC0050(this, 0);
+    s32 ret = false;
+
+    switch (this->behaviour) {
+        case ENNB_BEHAVIOUR_0:
+            if (!func_80BBFFD4(this, cutscene)) {
+                break;
+            }
+        // fallthrough
+        case ENNB_BEHAVIOUR_2:
+        case ENNB_BEHAVIOUR_4:
+        case ENNB_BEHAVIOUR_6:
+        case ENNB_BEHAVIOUR_8:
+            Camera_SetTargetActor(Play_GetCamera(play, ActorCutscene_GetCurrentSubCamId(cutscene)), &this->actor);
+            this->behaviour++;
+            ret = true;
+            break;
+
+        case ENNB_BEHAVIOUR_1:
+        case ENNB_BEHAVIOUR_3:
+        case ENNB_BEHAVIOUR_5:
+        case ENNB_BEHAVIOUR_7:
+            if ((this->actor.child != NULL) && (this->actor.child->update != NULL)) {
+                Camera_SetTargetActor(Play_GetCamera(play, ActorCutscene_GetCurrentSubCamId(cutscene)),
+                                      this->actor.child);
+            }
+            this->behaviour++;
+            ret = true;
+            break;
+
+        case ENNB_BEHAVIOUR_9:
+            ActorCutscene_Stop(cutscene);
+            this->behaviour++;
+            ret = true;
+            break;
+    }
+
+    return ret;
+}
+
+s32 func_80BC01DC(Actor* thisx, PlayState* play) {
+    EnNb* this = THIS;
+    s32 pad;
+    s32 ret = false;
+
+    switch (this->behaviour) {
+        case ENNB_BEHAVIOUR_0:
+            if (Player_GetMask(play) == PLAYER_MASK_ALL_NIGHT) {
+                this->behaviour = ENNB_BEHAVIOUR_1;
+            } else {
+                this->behaviour = ENNB_BEHAVIOUR_5;
+            }
+            break;
+
+        case ENNB_BEHAVIOUR_1:
+            // Setup a black fill-screen, although initialize to 0 alpha
+            Play_FillScreen(&play->state, true, 0, 0, 0, 0);
+            this->storyTimer = 40;
+            this->behaviour = (u16)(this->behaviour + 1);
```

Comment:
Strange that the other increments of `this->behaviour` only need `this->behaviour++`

Reply:
Yeah, this is very weird, I have no clue why only this needs this cast. 
Should we consider adding a `//! FAKE` comment?